ocaml/typing/types.mli

```diff
@@ -523,7 +523,10 @@ and record_representation =
      contains it and the tag of the relevant constructor of that variant. *)
   | Record_boxed of layout array
   | Record_float (* All fields are floats *)
-
+  | Record_ufloat
+  (* All fields are [float#]s.  Same runtime representation as [Record_float],
+     but operations on these (e.g., projection, update) work with unboxed floats
+     rather than boxed floats. *)
```

Comment:
I have hesitated to do this because we'll want to allow `any` in these declarations soon enough.  Though maybe even then a `sort option` would be better?  Possibly less convenient during inference, though.

Reply:
`sort option` seems about right, actually. That is, when we're checking, say, a field access, if the label has a sort, we don't need to check that it's representable. If it doesn't, then we'll need to check that the particular instantiation of the record in play at that point is representable. Storing an `option` communicates this nicely.

Actually, an even better idea: just leave a CR for v5 saying that it should be `sort option`. No reason to change all this code when we'll be doing so later anyway.